fab/operations/offline.py

```diff
@@ -0,0 +1,54 @@
+import os
+
+from fabric.api import env, local
+from fab.utils import generate_bower_command
+from fab.const import (
+    OFFLINE_STAGING_DIR,
+    WHEELS_ZIP_NAME,
+    HQ_ZIP_NAME,
+    BOWER_ZIP_NAME,
+    NPM_ZIP_NAME,
+)
+
+
+def prepare_zipfiles():
+    local(
+        'git clone --depth 1 --recursive https://github.com/dimagi/commcare-hq.git {}/commcare-hq'
+        .format(OFFLINE_STAGING_DIR)
+    )
+
+    # After we've created our HQ code zip, let's get bower and npm
```

Comment:
i've use [gitosis](https://git-scm.com/book/no-nb/v1/Git-on-the-Server-Gitosis) which looks like it just installs from a git repo which should be doable. only providing this information if useful, not advocating for changing what you have here since i'm sure you thought about this more than me